app/src/main/java/com/tari/android/wallet/ui/activity/SplashActivity.kt

```diff
@@ -46,15 +46,22 @@ class SplashActivity : BaseActivity() {
 
     override val contentViewId = R.layout.activity_splash
 
-    override fun onStart() {
-        super.onStart()
-        // splash is finished, move on to the main activity
-        val intent = Intent(this@SplashActivity, AuthActivity::class.java)
-        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
-        startActivity(intent)
-        // finish this activity
-        finish()
-    }
+    private val uiHandler = Handler()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
 
+        uiHandler.postDelayed({
+            val intent = Intent(this@SplashActivity, CreateWalletActivity::class.java)
+            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
+            startActivity(intent)
```

Comment:
You can turn all this into a function and call it in here on a WeakReference of the activity.

Reply:
As I have said earlier, reference won't be kept once we clean handler callbacks.